### PR TITLE
added test only function to return SignerCapabilityOfferProofChalleng…

### PR DESCRIPTION
**Pull request by dhruvja**
### Description
This PR is regarding this issue https://github.com/aptos-labs/aptos-core/issues/6109

I had created a PR before but it broke due to an update https://github.com/aptos-labs/aptos-core/pull/6110. Here is the new one.

### CODE REVIEW

**Review by alinush — APPROVED** [review]
LGTM!

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 1]
```diff
@@ -684,6 +684,25 @@ module aptos_framework::account {
...
+        let proof_challenge = SignerCapabilityOfferProofChallengeV2 {
+            sequence_number: borrow_global_mut<Account>(source_address).sequence_number,
+            source_address,
+            recipient_address,
+        };
+
+        return proof_challenge;
+    } 
+
+
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/account.move:705]:
```suggestion

```

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 2]
```diff
@@ -684,6 +684,25 @@ module aptos_framework::account {
...
+        source_address: address,
+        recipient_address: address
+    ) acquires Account {
+        let proof_challenge = SignerCapabilityOfferProofChallengeV2 {
+            sequence_number: borrow_global_mut<Account>(source_address).sequence_number,
+            source_address,
+            recipient_address,
+        };
+
+        return proof_challenge;
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/account.move:702]:
```suggestion
        SignerCapabilityOfferProofChallengeV2 {
            sequence_number: borrow_global_mut<Account>(source_address).sequence_number,
            source_address,
            recipient_address,
        }
```

**Comment by davidiw** [comment]
Looks good, it might need to update the move docs, let's find out.

**Review by davidiw — APPROVED** [review]

**Pull request merged**